packages/typespec-ts/src/transform/transformApiVersionInfo.ts

```diff
@@ -67,7 +67,7 @@ function getOperationQueryApiVersion(
       operationGroup
     );
     for (const op of operations) {
-      const route = ignoreDiagnostics(getHttpOperation(program, op));
+      const route = ignoreDiagnostics(getHttpOperationWithCache(program, op));
```

Comment:
Every time `getHttpOperation` would build a new http operation according to original one so we need to cache them otherwise the appending info would be lost.